### DISCUSSION TITLE
[@types/three] Missing subpath exports in package.json

### DISCUSSION THREAD

**Discussion by Exifers** [DefinitelyTyped/DefinitelyTyped#64681] [Issues with a @types package]
I'm trying to import [BufferGeometryUtils](https://threejs.org/docs/#examples/en/utils/BufferGeometryUtils), using [node16 module resolution](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#ecmascript-module-support-in-nodejs):

```typescript
import * as BufferGeometryUtils from "three/addons/utils/BufferGeometryUtils.js";
```

but it doesn't find the types :
```
Could not find a declaration file for module 'three/addons/utils/BufferGeometryUtils.js'. '/home/emmanuel/Documents/cube/node_modules/.pnpm/three@0.150.1/node_modules/three/examples/jsm/utils/BufferGeometryUtils.js' implicitly has an 'any' type.
  If the 'three' package actually exposes this module, consider sending a pull request to amend 'https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/three'
```

Also, the same import works well with plain nodejs (without typescript). :point_right:  [Here is a reproduction](https://github.com/Exifers/types-three-addon-import).

**Three package structure:** 

Three is using **package.json exports field** [here](https://github.com/mrdoob/three.js/blob/dev/package.json#L8), that remaps `./addons/*` to `./examples/jsm/*`, so `three/addons/utils/BufferGeometryUtils.js` is remapped to `three/examples/jsm/utils/BufferGeometryUtils.js` which exists so the javascript file is well found.

The corresponding types are in `@types/three/examples/jsm/utils/BufferGeometryUtils.d.ts` But **@types/three has no package.json exports**, so this file is not found.

**Shouldn't we add the package.json exports in @types/three too ?**

Like this for example :
```
   "types": "index.d.ts",
   "exports": {
      "./addons/*": "./examples/jsm/*"
    },
```

[a comment by an automated account is not shown]

**Comment by Methuselah96** [comment]
Thanks for bringing this up! You're absolutely right, we have added the `exports` field to the `package.json` in https://github.com/three-types/three-ts-types/pull/340. This change should be released with `@types/three@0.150.0`, we're just a little behind on getting that release out. Let me know if you have any further questions.

**Reply by Exifers** [reply]
Perfect thanks !
